pkg/api/store/cluster/cluster_store.go

```diff
@@ -88,3 +95,16 @@ func canUseClusterName(apiContext *types.APIContext, requestedName string) error
 
 	return nil
 }
+
+func setKubernetesVersion(data map[string]interface{}) {
+	rkeConfig, ok := values.GetValue(data, "rancherKubernetesEngineConfig")
```

Comment:
there is a way to get a nested value using values, I believe method is called GetValueN

Reply:
done